docs/api/http-v1-batch.md

```diff
@@ -241,12 +184,102 @@ a sample request and response that includes one missing object:
 >     }
 <   ]
 < }
+```
+
+Object error codes should match HTTP status codes where possible:
+
+* 404 - The object does not exist on the server.
+* 410 - The object was removed by the owner.
+* 422 - Validation error.
 
-The error codes are integers that map to similar HTTP status codes where
-possible. Specifics can be included by the server in the `message` property.
+Validation errors can only occur on `upload` requests. Servers must verify
+that OIDs are valid SHA-256 strings, and that sizes are positive integers.
+Servers should also set an upper bound for the allowed object size too. Here's a
```

Comment:
Should this be "Servers *can* also set an upper bound"? I don't think an upper bound would be commonplace for self-hosting, only when you're providing hosting for other people.